ibis/expr/types/arrays.py

```diff
@@ -467,26 +507,28 @@ def map(self, func: Deferred | Callable[[ir.Value], ir.Value]) -> ir.ArrayValue:
         │ [6]                    │
         │ []                     │
         └────────────────────────┘
-        """
-        if isinstance(func, Deferred):
-            name = "_"
-            resolve = func.resolve
-        elif callable(func):
-            name = next(iter(inspect.signature(func).parameters.keys()))
-            resolve = func
-        else:
-            raise TypeError(
-                f"`func` must be a Deferred or Callable, got `{type(func).__name__}`"
-            )
 
-        parameter = ops.Argument(
-            name=name, shape=self.op().shape, dtype=self.type().value_type
-        )
-        body = resolve(parameter.to_expr())
-        return ops.ArrayMap(self, param=parameter.param, body=body).to_expr()
+        You can optionally include a second index argument in map function
```

Comment:
Typo, guessing you meant something like

```suggestion
        You can optionally include a second index argument in the mapped function
```